src/include/interval_set.h

```diff
@@ -674,10 +675,9 @@ class interval_set {
     if (big.size() / size() < 10)
       return subset_size_sym(big);
 
-    for (typename Map::const_iterator i = m.begin();
-         i != m.end();
-         i++) 
-      if (!big.contains(i->first, i->second)) return false;
+    for (auto it = m.begin(); it != m.end(); ++it) {
```

Comment:
i found you are switching to the range-based loop and structured binding in `std::ostream& operator<<()`. does it make sense to change this for loop as well?